R/statistics.R

```diff
@@ -691,3 +691,247 @@ SecretarySampler <- R6::R6Class("SecretarySampler", public = list(
     }
 )
 )
+#' Create an abstract streamer for the multi-armed bandit problem.
+#'
+#' @description \code{MultiArmedBandit}
+#'
+#' @docType class
+#'
+#'
+#' @export
+#' @format An \code{\link{R6Class}} generator object
+MultiArmedBandit <- R6::R6Class(
+    "MultiArmedBandit",
+    public = list(
+        #' @description Creates a new \code{MultiArmedBandit} streamer object.
+        #'
+        #' @param K the number of arms
+        #' @param T number of rounds / draws
+        #'
+        #' @return The new \code{MultiArmedBandit} (invisibly)
+        initialize = function(K, T, N = NULL) {
+            private$K <- K
+            private$T <- T
+            arm_info <- list()
+            for (k in seq(K)) {
+                arm_info[[k]] <- list(mean = CMA$new())
+            }
+            private$arm_info <- arm_info
+            invisible(self)
+        },
+        #' @description Update the \code{MultiArmedBandit} streamer object.
+        #'
+        #' @param x a single draw from an arm
+        #' @param k index of the chosen arm
+        #'
+        #'
+        #' @return The updated code{MultiArmedBandit} (invisibly)
+        update = function(x, k) {
+            if (length(x) > 1) {
+                stop("Only single-value updated allowed")
+            }
+            private$arm_info[[k]]$mean$update(x)
+            private$n_observed <- private$n_observed + 1
+        }
+    ),
+    active = list(
+        #' @description Returns the the list containing values known to
+        #' to the Bandit in the current state.
+        #'
+        #'
+        #' @return list with summary of the state of the Bandit.
+        value = function() {
```

Comment:
I think we said that `value` would just return the next arm to pull. `summary` can be used for the full state.